pkg/services/ngalert/schedule/schedule_test.go

```diff
@@ -106,7 +105,7 @@ func TestWarmStateCache(t *testing.T) {
 		Labels:            labels,
 	}
 	_ = dbstore.SaveAlertInstances(ctx, instance2)
-	st := state.NewManager(testMetrics.GetStateMetrics(), nil, dbstore, &image.NoopImageService{}, clock.NewMock(), &state.FakeHistorian{})
+	st := state.NewManager(testMetrics.GetStateMetrics(), nil, dbstore, &state.NoopImageService{}, clock.NewMock(), &state.FakeHistorian{})
```

Comment:
This is strange, I thought the goal was the break dependency between these two packages, but we have `state.NoopImageService`, what are the alternatives tried?

Reply:
The only references to `NoopImageService` in the entire codebase are the `state` tests. It's used as a test fake, in tests that are asserting on the way that `state` invokes the image functionality. I feel that `state` is therefore the right place for it, as it's a utility to support `state`'s tests.

To put it another way - there are no `image` tests that use this testing fake. Why keep it in the `image` package if it does not support `image` tests?

Reply:
It will be used in prod code of backtesting as well (see https://github.com/grafana/grafana/pull/57318)

Reply:
I will move mock and both fakes to a testing sub-package off state in a future PR. There is not currently a sub-package for state mocks (only `eval` uses this pattern). The sub-package for mocks is a great idea IMO, I just think creating new package probably implies a new PR scope.